src/pybind/mgr/orchestrator/module.py

```diff
@@ -571,8 +571,8 @@ def _mds_add(self, fs_name, placement=None):
 
     @_cli_write_command(
         'orch daemon add rgw',
-        'name=realm_name,type=CephString '
-        'name=zone_name,type=CephString '
+        'name=realm_name,type=CephString,req=false '
+        'name=zone_name,type=CephString,req=false '
```

Comment:
this change seems a bit unrelated to this PR?

Reply:
Yes and no, this means we can pass: ` ceph orch add rgw -i <json>` with out having to pass in the realm and zone on the command line.

Now that, or rather when, we can use the config-key store then maybe the `-i` becomes out of scope for the PR.